src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier randomColor = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figures [] figures = Figures.values();
+        int index = random.nextInt(figures.length);
+        //Figures randomFigure = figures [index];
+        switch (figures[index]) {
+            case CIRCLE:
+                return new Circle(randomColor.getRandomColor(),Figures.CIRCLE,random.nextInt(100));
```

Comment:
1. Add private methods (like getRandomCircle) for each type
2. https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced
![image](https://user-images.githubusercontent.com/20326858/214279113-10266018-4dfb-4246-b5fa-96b5dea012fb.png)